sklearn/preprocessing/tests/test_data.py

```diff
@@ -1952,6 +1955,98 @@ def test_one_hot_encoder_unknown_transform():
     assert_raises(ValueError, oh.transform, y)
 
 
+def check_categorical(X):
+    enc = CategoricalEncoder()
+    Xtr1 = enc.fit_transform(X)
+
+    enc = CategoricalEncoder()
+    Xtr2 = enc.fit_transform(X)
+
+    enc = CategoricalEncoder(sparse=False)
+    Xtr3 = enc.fit_transform(X)
+
+    assert_allclose(Xtr1.toarray(), Xtr2.toarray())
+    assert_allclose(Xtr1.toarray(), Xtr3)
+
+    assert sparse.issparse(Xtr1)
+    assert sparse.issparse(Xtr2)
+    return Xtr1.toarray()
+
+
+def test_categorical_encoder():
+    X = [['abc', 1, 55], ['def', 2, 55]]
+
+    Xtr = check_categorical(np.array(X)[:, [0]])
+    assert_allclose(Xtr, [[1, 0], [0, 1]])
+
+    Xtr = check_categorical(np.array(X)[:, [0, 1]])
+    assert_allclose(Xtr, [[1, 0, 1, 0], [0, 1, 0, 1]])
+
+    Xtr = CategoricalEncoder().fit_transform(X)
+    assert_allclose(Xtr.toarray(), [[1, 0, 1, 0,  1], [0, 1, 0, 1, 1]])
```

Comment:
We should probably explain the handling of constant features. Does this make the most sense? Not sure what else....